torch/csrc/lazy/core/tensor_util.h

```diff
@@ -0,0 +1,68 @@
+#pragma once
+
+#include <torch/csrc/lazy/backend/backend_interface.h>
+#include <torch/csrc/lazy/core/shape.h>
+
+#include <string>
+#include <vector>
+
+namespace torch {
+namespace lazy {
+
+TORCH_API std::vector<int64_t> ComputeArrayStrides(
+    c10::ArrayRef<int64_t> sizes);
+
+TORCH_API std::vector<at::Tensor> DataHandlesToTensors(
+    c10::ArrayRef<BackendDataPtr> data_handles,
+    at::ScalarType dest_element_type);
+
+// Uploads an ATEN tensor data to the device and fetches the corresponding
+// device data handle.
+TORCH_API BackendDataPtr
+TensorToDataHandle(const at::Tensor& tensor, const BackendDevice& device);
+
+// Retrieves the device data handles by parallel uploading data onto the
+// corresponding devices.
+TORCH_API std::vector<BackendDataPtr> CreateTensorsData(
+    const std::vector<at::Tensor>& tensors,
+    const std::vector<BackendDevice>& devices);
+
+// Makes a deep copy of an ATEN tensor.
+static inline at::Tensor CopyTensor(const at::Tensor& ref) {
```

Comment:
I don't think you need `static inline` in functions defined in headers. Is it coding style?

Reply:
I have seen both `static inline` and `inline` get used in core. There is actually a lint warning for `static`, so this might be a good time to remove it.

Reply:
Just for my education, what are the benefits/effects of using static in non-member functions in headers? For `inline`, functions defined in header should be implicitly `inline` unless I missed something.